libraries/botbuilder/src/teamsInfo.ts

```diff
@@ -155,13 +159,33 @@ export class TeamsInfo {
             },
             activity: activity,
         } as ConversationParameters;
-        const connectorClient = (context.adapter as BotFrameworkAdapter).createConnectorClient(
-            context.activity.serviceUrl
-        );
-        const conversationResourceResponse = await connectorClient.conversations.createConversation(convoParams);
-        const conversationReference = TurnContext.getConversationReference(context.activity);
-        conversationReference.conversation.id = conversationResourceResponse.id;
-        return [conversationReference as ConversationReference, conversationResourceResponse.activityId];
+
+        let conversationReference: Partial<ConversationReference>;
+        let newActivityId: string;
+
+        if (botAppId && context.adapter instanceof CloudAdapter) {
```

Comment:
I'm inclined to say that `instanceof` checks within the same package are ok since they won't be affected by cross-package version inconsistencies. Thoughts @stevengum?

Reply:
I think it's fine to do an `instanceof` check with the caveat that someone subclassing the `BotFrameworkAdapter` could implement their own `createConversationAsync()` which would likely not be used as expected.

I think it's fine because it's a new API and we're planning on shifting the public focus to the `CloudAdapter` with the intent of deprecating the `BotFrameworkAdapter` class.

Reply:
Resolved?

Reply:
Yeah, we can always walk this back and move back to `catch (err) { if (err.message === "NotImplemented") { ... } }`